src/Polar/kernels.jl

```diff
@@ -204,31 +205,92 @@ KA.@kernel function adj_residual_node_kernel!(F, adj_F, vm, adj_vm, va, adj_va,
     end
 end
 
+KA.@kernel function adj_residual_node_kernel1!(adj_vm, adj_va,
+                                  colptr, rowval,
+                                  edge_vm_from, edge_va_from) 
+    i = @index(Global, Linear)
+    @inbounds for c in colptr[i]:colptr[i+1]-1
+        adj_vm[i] += edge_vm_from[c]
+        adj_va[i] += edge_va_from[c]
+    end
+end
+
+KA.@kernel function adj_residual_node_kernel2!(adj_vm, adj_va, 
```

Comment:
Interesting. Is `adj_residual_node_kernel2!` faster than `adj_residual_node_kernel1!`?

Reply:
In fact, I am not sure to understand what exactly the purpose of `adj_residual_node_kernel2` is

Reply:
kernel1 adds the from-edge to the nodes and goes over all buses. kernel2 goes over all the to-edges and is a bit trickier, but now I realize it may still be wrong. It should also go over all the buses. Not sure why each thread gets a line.

Reply:
I merged them.